mypy/messages.py

```diff
@@ -881,8 +881,10 @@ def format_string_list(s: Iterable[str]) -> str:
     assert len(l) > 0
     if len(l) == 1:
         return l[0]
-    else:
+    elif len(l) <= 5:
         return '%s and %s' % (', '.join(l[:-1]), l[-1])
+    else:
+        return '%s, ... and %s (%i methods suppressed)' % (', '.join(l[:4]), l[-1], len(l) - 5)
```

Comment:
I think it's better to show the first 2 and the last 2, or even the first 2 and the last one, so that the suppressed count is always >= 2 (or >= 3).